### PR TITLE
[PACT-607] Added "place of deposit" to record edit page.

### PR DESCRIPTION
**Pull request by tna-erasmos**

### CODE REVIEW

**Review by rwalpole — APPROVED** [review]

File: app/uk/gov/nationalarchives/omega/editorial/views/editSetRecordEdit.scala.html [hunk 1]
```diff
@@ -159,21 +164,47 @@
...
+    name = "placeOfDeposit",
+    items = {
+        val noSelectionAsMap = Map("" -> messages("edit-set.record.edit.place-of-deposit.no-selection"))
+        val corporateBodiesAsMap = noSelectionAsMap ++ corporateBodies.map(corporateBody => (corporateBody.id, corporateBody.name)).toMap
+        corporateBodiesAsMap
+                .map { case (id, label) =>
+            SelectItem(
+                selected = editSetRecordForm.data.get("placeOfDeposit").contains(id),
+                text = label,
+                value = Some(id).filter(_.nonEmpty),
```
Comment (felixgb) [inline app/uk/gov/nationalarchives/omega/editorial/views/editSetRecordEdit.scala.html:195]: Isn't this `id` always going to be "placeOfDeposit"? Why the filter?

File: test/controllers/EditSetControllerSpec.scala [hunk 2]
```diff
@@ -244,72 +291,107 @@ class EditSetControllerSpec extends BaseSpec {
...
+
+            val result = submitWhileLoggedIn(1, "COAL.2022.V5RJW.P", values)
 
-            status(editRecordPage) mustBe BAD_REQUEST
-            val document = asDocument(editRecordPage)
-            document must haveHeading(s"TNA reference: $ccrToAssert")
+            status(result) mustBe BAD_REQUEST
+            assertPageAsExpected(
+              asDocument(result),
+              ExpectedPage(
```
Comment (felixgb) [inline test/controllers/EditSetControllerSpec.scala:313]: Do you think it's worth having a default expected page case class instance, with valid values, at the top level of this spec? Then tests expecting invalid values could `.copy(....` the properties that are being asserted for that spec.

The bonuses that I see would be:
* cut down on noise in tests
* make it explicit what kind of error message / value is being asserted.

File: app/uk/gov/nationalarchives/omega/editorial/views/editSetRecordEdit.scala.html (hunk 1 above)
Comment (tna-erasmos) [reply app/uk/gov/nationalarchives/omega/editorial/views/editSetRecordEdit.scala.html:195]: `id` is the key of the map; when it should default to the blank option, I make the value a `None`.

File: test/controllers/EditSetControllerSpec.scala (hunk 2 above)
Comment (tna-erasmos) [reply test/controllers/EditSetControllerSpec.scala:313]: I prefer tests which are explicit to all expected values; so I don't see it as noise.

**Review by felixgb — APPROVED** [review]

File: app/uk/gov/nationalarchives/omega/editorial/views/editSetRecordEdit.scala.html [hunk 3]
```diff
@@ -159,21 +164,47 @@
...
+                disabled = id == ""
             )
+        }.toSeq
+    },
+    label = Label(forAttr = Some("placeOfDeposit"), content = Text(messages("edit-set.record.edit.place-of-deposit"))),
```
Comment (userjd) [inline app/uk/gov/nationalarchives/omega/editorial/views/editSetRecordEdit.scala.html:200]: @tna-erasmos should the error message for held by display the generic message and the default option in the dropdown show this message `edit-set.record.edit.place-of-deposit`

> You must choose an option

File: conf/messages [hunk 4]
```diff
@@ -13,17 +13,18 @@ edit-set.record.edit.heading=TNA reference: {0}
 edit-set.record.edit.id=PAC-ID: {0}
 edit-set.record.edit.legal-status=Legal Status
 edit-set.record.edit.legend=Intellectual properties
+edit-set.record.edit.place-of-deposit.no-selection=You must choose an option
+edit-set.record.edit.place-of-deposit=Held by
```
[the remark this answers is not in the document]
Comment (tna-erasmos) [reply conf/messages:17]: Yes, but I'd like to confirm whether we should use the same message, as it doesn't indicate the field in error - which looks odd in the summary.

[the remark this answers is not in the document]
Comment (tna-erasmos) [reply conf/messages:17]: On second thought, I'll change it for now.

File: app/uk/gov/nationalarchives/omega/editorial/views/editSetRecordEdit.scala.html (hunk 3 above)
Comment (tna-erasmos) [reply app/uk/gov/nationalarchives/omega/editorial/views/editSetRecordEdit.scala.html:200]: I'll change it.

Comment (tna-erasmos) [reply app/uk/gov/nationalarchives/omega/editorial/views/editSetRecordEdit.scala.html:200]: Changed.

File: conf/messages (hunk 4 above)
[the remark this answers is not in the document]
Comment (tna-erasmos) [reply conf/messages:16]: Changed.

[the remark this answers is not in the document]
Comment (userjd) [reply conf/messages:16]: LGTM

File: app/uk/gov/nationalarchives/omega/editorial/views/editSetRecordEdit.scala.html (hunk 3 above)
Comment (userjd) [reply app/uk/gov/nationalarchives/omega/editorial/views/editSetRecordEdit.scala.html:200]: LGTM

**Review by userjd — APPROVED** [review]

**Pull request merged**